File: packages/agent-sdk-python/examples/run_agent.py
import os
import sys

from magicpot_agent_sdk import AgentRunRequest, HttpAgentTransport, MagicAgentClient

base_url = os.environ.get("MAGICPOT_SDK_URL")
token = os.environ.get("MAGICPOT_SDK_TOKEN")
if not base_url or not token:
    raise RuntimeError("Set MAGICPOT_SDK_URL and MAGICPOT_SDK_TOKEN.")

client = MagicAgentClient(HttpAgentTransport(base_url, token))
result = client.run(
    AgentRunRequest(
        os.environ.get("MAGICPOT_AGENT_ID", "default"),
        {"prompt": " ".join(sys.argv[1:]) or "Hello from the Python SDK"},
    )
)
print(result)

File: packages/agent-sdk-typescript/examples/run-agent.ts
import { HttpAgentTransport, MagicAgentClient } from '../src/index.js'

const baseUrl = process.env.MAGICPOT_SDK_URL
const token = process.env.MAGICPOT_SDK_TOKEN
if (!baseUrl || !token) throw new Error('Set MAGICPOT_SDK_URL and MAGICPOT_SDK_TOKEN.')

const client = new MagicAgentClient(new HttpAgentTransport({ baseUrl, token }))
const result = await client.run({
  agentId: process.env.MAGICPOT_AGENT_ID ?? 'default',
  input: { prompt: process.argv.slice(2).join(' ') || 'Hello from the TypeScript SDK' }
})
console.log(JSON.stringify(result, null, 2))
